parse-hamnosys.py

```diff
@@ -153,122 +153,75 @@ def main(args):
     data["Dominant - Handshape - Thumb position"] = 0
     data["Dominant - Handshape - Bending"] = 0
 
-    for index, row in data.iterrows():
-        # Search for symmetry operators that consists of 3 symbols,
+    for index, _ in data.iterrows():
+        # Search for symmetry operators that consists of from 3 to 1 symbol,
         # remove if found
-        char = Hamnosys_copy[index][0:3]
-        if ((char == SymmetryOperatorsDict["1"]) or
-                (char == SymmetryOperatorsDict["9"])):
-            data.at[index, "Symmetry operator"] = 1
-            Hamnosys_copy[index] = Hamnosys_copy[index][3:]
-            continue
-        elif ((char == SymmetryOperatorsDict["2"]) or
-              (char == SymmetryOperatorsDict["10"])):
-            data.at[index, "Symmetry operator"] = 2
-            Hamnosys_copy[index] = Hamnosys_copy[index][3:]
-            continue
-
-        # Search for symmetry operators that consists of 2 symbols,
-        # remove if found
-        char = Hamnosys_copy[index][0:2]
-        if char == SymmetryOperatorsDict["3"]:
-            data.at[index, "Symmetry operator"] = 3
-            Hamnosys_copy[index] = Hamnosys_copy[index][2:]
-            continue
-        elif char == SymmetryOperatorsDict["4"]:
-            data.at[index, "Symmetry operator"] = 4
-            Hamnosys_copy[index] = Hamnosys_copy[index][2:]
-            continue
-        elif char == SymmetryOperatorsDict["5"]:
-            data.at[index, "Symmetry operator"] = 5
-            Hamnosys_copy[index] = Hamnosys_copy[index][2:]
-            continue
-        elif char == SymmetryOperatorsDict["6"]:
-            data.at[index, "Symmetry operator"] = 6
-            Hamnosys_copy[index] = Hamnosys_copy[index][2:]
-            continue
-
-        # Search for symmetry operators that consists of 1 symbol,
-        # remove if found
-        char = Hamnosys_copy[index][0]
-        if char == SymmetryOperatorsDict["7"]:
-            data.at[index, "Symmetry operator"] = 7
-            Hamnosys_copy[index] = Hamnosys_copy[index][1:]
-            continue
-        elif char == SymmetryOperatorsDict["8"]:
-            data.at[index, "Symmetry operator"] = 8
-            Hamnosys_copy[index] = Hamnosys_copy[index][1:]
-            continue
-        # Check if non dominan hand is analyzed as only and set flag
-        elif char == "":
-            data.at[index, "Symmetry operator"] = 0
-            Hamnosys_copy[index] = Hamnosys_copy[index][1:]
-            data.at[index, "NonDom first"] = 1
-            continue
+        for i in range(1, 10):
+            j = 3 if i < 3 else (2 if i < 7 else 1)
+            char = Hamnosys_copy[index][0:j]
+            if set(char) == set(SymmetryOperatorsDict[str(i)]):
+                data.at[index, "Symmetry operator"] = i % 9
+                Hamnosys_copy[index] = Hamnosys_copy[index][j:]
+                continue
 
```

Comment:
@mplantykow look here - I just shorten all logic for Symmetry in one loop - range 1-9 so for all 9 HamNoSys signs

Reply:
9 is not a symmetry operator. Before your change was introduced 9 was analyzed in a separate if and if found 1 was assigned to column named "NonDom first", not to symmetry operators.